### ISSUE TITLE
Build anaconda-project -> conda-project conversion capability

### ISSUE THREAD

**Issue by AlbertDeFusco** [conda-incubator/conda-project#62]
I *may* not want to have a backwards-compatible runtime, but instead offer a way to translate the `anaconda-project.yml`†  to appropriate conda-project files. We would need to setup a minimum level of parity between the two. One of the topics not yet considered here is jinja2 templating in commands.

†It may be possible to attempt a direct translation of anaconda-project-lock.yml to conda-lock.yml, but it may require reaching out the repo to retrieve certain metadata (like hashes) that anaconda-project-lock.yml does not capture.

**Comment by jbednar** [comment]
Same!

**Comment by jbednar** [comment]
conda-project will still support having it all in one file, right? Supporting a separate environment file is great in some cases, but in others I don't want a bunch of confusing files in my project directory, with just one to handle all the above. Assuming it's supported, I'd hope that a conversion script generates just one file, so that it is much easier to compare before/after conversion to spot any issues.

Whether to make it automatic is up for debate; my vote would be to support read-only automatic conversion, i.e., reading anaconda-project.yml and respecting it, but not trying to add packages, etc. to it when the user invokes such an option.

**Comment by AlbertDeFusco** [comment]
Yeah, I'm thinking I'm not against a one-file approach for conda-project, but it is not currently implemented.

Would this sound reasonable? It would allow external file and in-project-file dependencies.

```yaml
# conda-project.yml
name: proj

environments:
  default:
   - dependencies:
       - <pkg>
     channels:
       - defaults
   - ./dev-extras.yml
```

**Comment by mattkram** [comment]
Would this entail migrating your conversation script to maybe a generic `convert` subcommand?

E.g.

```shell
conda project convert --from anaconda-project anaconda-project.yml
```

Maybe `import` instead?

**Comment by AlbertDeFusco** [comment]
Let's go do that. And make `--from anaconda-project` auto select if an anaconda-project.yml file is passed, or found in the current working dir?

**Comment by mattkram** [comment]
👍 I agree. Have an explicit option, and handle that one with an implicit default.

We can use the same subcommand in future for e.g. notebooks, or other things. There's another issue for that.

**Comment by mattkram** [comment]
I'm not sure it should auto-find the file. We should require the argument. But we can infer the source type from the filename.
